pkg/cmd/opts/deprecations.go

```diff
@@ -0,0 +1,14 @@
+package opts
+
+import (
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/jenkins-x/jx/pkg/util"
+)
+
+// LogInstallDeprecated logs a warning message about deprecation
+func LogInstallDeprecated() {
```

Comment:
I would make this function more generic, that other commands can use it. e.g. provide the `recommended command` as an argument and also the link to the docs.

Reply:
also `Cobra Command` seems to have a `Deprecated` flag which can be set on the command.

```
type Command struct {
	// Deprecated defines, if this command is deprecated and should print this string when used.
	Deprecated string
```

which prints this message before the command is executed:

```Go
func (c *Command) execute(a []string) (err error) {
	if c == nil {
		return fmt.Errorf("Called Execute() on a nil Command")
	}

	if len(c.Deprecated) > 0 {
		c.Printf("Command %q is deprecated, %s\n", c.Name(), c.Deprecated)
	}
```

I am wondering if we can leverage this instead.